### ISSUE TITLE
exp: Train diffusion module to work from distogram rather than pair representation

### ISSUE THREAD

**Issue by timodonnell** [Open-Athena/helico#9]
## Question
Can we train a diffusion module to work from the distogram rather than the pair representation? How much accuracy is lost by doing this?

## Hypothesis
The diffusion module can be retrained to take (in addition to all the information it already takes), the distogram output rather than the final pair representation.

## Background
The pair features outputted by the AF3 trunk can be transformed to a distogram by a single dense layer applied to each (i,j) pair that cuts the channel dimension in half. The trunk is supervised on this distogram output, but it is not provided to the diffusion module. We know the pair → distogram operation loses information, but we don’t know how important that lost information is to the diffusion model. The goal of this experiment is to understand this. 

## Approach
* Start from existing weights for the whole model
* Rewire the diffusion conditioning information to substitute the distogram output rather than the pair features
* Retrain the model. Freeze the weights of the trunk, including the distogram output layer. Allow the weights of the diffusion module to change
* May want to consider an initial burn in period where only the first few layers that of the diffusion module that interact with the pair features are allowed to change
* We want to try to do this on a single 8x GPU node. We want to use as large a batch size as possible for the diffusion model outputs. To free up space for this, at each step we should run the trunk once to get the distogram and other outputs, then make sure the intermediate tensors generated by the trunk run are no longer in memory when we do the forward and backward passes through the diffusion module.

Possible extensions (followups)
* We could try allowing the trunk weights to change well. May be worth using a much smaller learning rate for the trunk weights than the diffusion module weights.

## Compute estimate
Let’s try this for 10,000 steps on an 8x H100 or similar. At 5 seconds per step that should take about 14 hours.

## Success criteria
The best outcome would be that the model trains to nearly the same level of accuracy as the original model

For evals let's focus on protein monomers for now.

## Baselines

Compare against the baseline model using Protenix v1 weights

## Notes

Additional context for this series of experiments: In theory, the diffusion output of AF3 should very naturally allow it to sample multiple conformations (as opposed to the regression output used in AF2). In practice, however, the diffusion conditioning information created by the trunk (the final single and pair representations) almost overly constrains the diffusion model, causing it to mostly sample the same conformation at each sample (especially when an MSA is available). This also makes guidance methods hard to apply in the diffusion module since guidance tends to conflict with the fixed conformation outputted by the trunk. The goal in these experiments is to create a better model (e.g. more diverse conformations, more guidable diffusion outputs) by bottlenecking the conditioning information at the trunk / diffusion module interface.
